components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/core/ServerIdpManagementService.java

```diff
@@ -110,22 +110,26 @@
     /**
      * Get list of identity providers.
      *
-     * @param attributes Required attributes in the IDP list response.
+     * @param requiredAttributes Required attributes in the IDP list response.
      * @param limit      Items per page.
      * @param offset     Offset.
      * @param filter     Filter string. E.g. filter="name" sw "google" and "isEnabled" eq "true"
      * @param sortBy     Attribute to sort the IDPs by. E.g. name
      * @param sortOrder  Order in which IDPs should be sorted. Can be either ASC or DESC.
      * @return IdentityProviderListResponse.
      */
-    public IdentityProviderListResponse getIDPs(String attributes, Integer limit, Integer offset, String filter, String
-            sortBy, String sortOrder) {
+    public IdentityProviderListResponse getIDPs(String requiredAttributes, Integer limit, Integer offset, String filter,
+                                                String sortBy, String sortOrder) {
 
-        handleNotImplementedCapabilities(attributes);
         try {
+            List<String> requestedAttributeList = null;
+            if (StringUtils.isNotBlank(requiredAttributes)) {
+                requestedAttributeList = new ArrayList<>(Arrays.asList(requiredAttributes.split(",")));
```

Comment:
we can use StringUtils.split and void if (StringUtils.isNotBlank(requiredAttributes))  check

Reply:
StringUtils.split(requiredAttributes, ",") will return null if requiredAttributes is null. 
Arrays.asList(StringUtils.split(requiredAttributes, ",")) will thow NPE if StringUtils.split(requiredAttributes, ",") return null, so we need to add null check.